ash/src/entry.rs

```diff
@@ -214,9 +230,11 @@ impl Entry {
     /// <https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/vkCreateInstance.html>
     ///
     /// # Safety
-    /// In order for the created [`Instance`] to be valid for the duration of its
-    /// usage, the [`Entry`](Self) this was called on must be dropped later than the
-    /// resulting [`Instance`].
+    ///
+    /// The resulting [`Instance`] **must** be destroyed through [`Instance::destroy_instance()`]
+    /// (_not_ [dropped][drop()]) before the [`Entry`] this was called on is [dropped][drop()], in
+    /// the broad sense of the word. That is, valid Vulkan usage requires the [`Instance`] to be
+    /// destroyed, and it otherwise becomes invalid to use after this [`Entry`] is dropped.
```

Comment:
- What is the broad sense of "dropped"?
- Instances do not have to be destroyed; rather, it is illegal (in general) to destroy a parent object without first destroying its children.
- It's not specifically invalid to use a non-destroyed instance after the `Entry` was dropped; rather, you have already invoked undefined behavior at that point and all bets are off.

Reply:
Further, dropping a `linked` `Entry` is harmless.

Reply:
> * What is the broad sense of "dropped"?

As explained above, referring to `**must**` instead of `dropped`. Will find a different way of writing that up.

> * Instances do not have to be destroyed; rather, it is illegal (in general) to destroy a parent object without first destroying its children.

I'll see how this can be worded in here: clearly stating that there is a parent/child relation here, and who those parent(s) are.

> * It's not specifically invalid to use a non-destroyed instance after the `Entry` was dropped; rather, you have already invoked undefined behavior at that point and all bets are off.

It is useful to mention this though. Perhaps drop the "valid Vulkan usage requires ... to be destroyed" bit? Especially considering the option below:

> Further, dropping a `linked` `Entry` is harmless.

I had considered this case but not thought how to best describe it here. Dropping it in that instance simply removes access to the function pointers inside it, but doesn't violate other pointers in any way.

Reply:
> I had considered this case but not thought how to best describe it here.

Perhaps something like "The instance, and any objects derived from it, may only be manipulated for the period described in constructor used to obtain this `Entry`"?

Reply:
Reworded this to state that the returned `Instance` object (and anything derived from it) should not be used anymore after `Entry` was dropped, unless created via `Entry::linked()` or `Entry::from_parts_1_1()`. Not entirely sure about the last one, since it has more complex library/lifetime requirements provided by the user...

Reply:
> Not entirely sure about the last one, since it has more complex library/lifetime requirements provided by the user...

Seems fine. No matter what else might be going on, the `Entry`'s lifetime isn't specifically relevant.

Reply:
It'd be too complex to write down, if we do it should be described on `from_parts_1_1()` instead which currently carries no documentation whatsoever. There's not `Arc<libloading::Library>` involved for the function-pointer lifetime at least.